### ISSUE TITLE
tuple of tuple arguments not allowed in parfor loop

### ISSUE THREAD

**Issue by albop** [numba/numba#9098]
The following code

import numpy as np
import numba as nb
from numba import prange

@nb.jit
def my_fun(vals, grid, coef, x):
    return vals[0] + coef[0] + x[0] # + grid[0][0]

@nb.jit(nopython=True, parallel=True)
def run_in_pfor(vals, grid, coef, x):
    t = 0.0
    for h in prange(6):
        res = my_fun(vals, grid, coef, x)
        t += res
    return t

vals = np.random.rand(5)
grid = ((0., 1., 5),)
coef = vals.copy()
x = np.array([0.22])
run_in_pfor(vals, grid, coef, x)
raises an exception ending in

```
numba.core.errors.LoweringError: Failed in nopython mode pipeline (step: native parfor lowering)
scalar type Tuple(float64, float64, int64) given for non scalar argument #3

File "notuple.py", line 12:
def run_in_pfor(vals, grid, coef, x):
    <source elided>
    t = 0.0
    for h in prange(6):
    ^

During: lowering "id=0[LoopNest(index_variable = parfor_index.6, range = (0, $const20.3, 1))]{40: <ir.Block at /home/pablo/Econforge/interpolation.py/notuple.py (12)>}Var(parfor_index.6, notuple.py:12)" at /home/pablo/Econforge/interpolation.py/notuple.py (12)
```
full stacktrace attached : [stacktrace.txt](https://github.com/numba/numba/files/12181997/stacktrace.txt) .

If I replace `grid = ((0., 1., 5),)` by `grid = (0., 1., 5)` it works (it is not used by the function anyway).

**Comment by albop** [comment]
Original discussion here: https://github.com/EconForge/interpolation.py/issues/107#issuecomment-1652042641

**Comment by guilhermeleobas** [comment]
Hi @albop, thanks for the report. I can confirm the bug in Numba and only works if parallel=True.

**Comment by albop** [comment]
Fair enough. Thanks for the clarification @DrTodd13 . I can certainly find an ugly workaround in the meantime ;-)
